### PR TITLE
763 rmc 115 generate initial

### PR DESCRIPTION
**Pull request by LukeLoze1**
# Motivation and Context

Shared templates for England, Wales and NI

# What has changed
Deleted old templates and mapping, added new ones.  Added new packcode column, this is populated when the ActionRequests are first processed on to the database, they are then referenced in the templates, previously were hardcoded

# How to test?
mvn clean install
run with ATs . with correct actionscheduler installed

# Links
https://trello.com/c/ZmGZEf9A/763-rmc-115-action-exporter-to-generate-initial-contact-print-files-13
https://github.com/ONSdigital/census-rm-action-scheduler/pull/11

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/action/export/service/ActionExportService.java [hunk 1]
```diff
@@ -27,11 +28,21 @@
...
 
+  private final HashMap<String, String> actionTypeToPackCodeMap =
+      new HashMap<String, String>() {
+        {
+          put("ICHHQE", "P_IC_H1");
+          put("ICHHQW", "P_IC_H2");
+          put("ICHHQN", "P_IC_H4");
+          put("ICL1E", "P_IC_ICL1");
+          put("ICL2W", "P_IC_ICL2");
+          put("ICL4E", "P_IC_ICL4");
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/ctp/response/action/export/service/ActionExportService.java]: Change `ICL4E` to `ICL4N`

File: src/main/resources/database/changes/release-22/add_contact_details_in_wales_template.sql [hunk 2]
```diff
@@ -43,4 +43,7 @@ VALUES
 
 
 DELETE FROM actionexporter.templatemapping WHERE actiontypenamepk = 'ICL2E';
-DELETE FROM actionexporter.template WHERE templatenamepk = 'icl2e'
\ No newline at end of file
+DELETE FROM actionexporter.template WHERE templatenamepk = 'icl2e'
+
+
+
```
Comment (NickGrantX) [inline src/main/resources/database/changes/release-22/add_contact_details_in_wales_template.sql:49]: why the blank lines?

File: src/main/resources/database/changes/release-23/initial_contact_and_questionaire_template.sql [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+INSERT INTO actionexporter.templatemapping
+(actiontypenamepk, templatenamefk, filenameprefix, datemodified, requesttype)
+VALUES
+('ICL2W','initial_contact', 'P_IC_ICL2', now(), 'print_service');
+
+-- Nireland
+INSERT INTO actionexporter.templatemapping
+(actiontypenamepk, templatenamefk, filenameprefix, datemodified, requesttype)
+VALUES
+('ICL4E','initial_contact', 'P_IC_ICL4', now(), 'print_service');
```
Comment (NickGrantX) [inline src/main/resources/database/changes/release-23/initial_contact_and_questionaire_template.sql]: Change to `ICL4N`

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Some `ICL4E`s need changing to `ICL4N`

File: src/test/java/uk/gov/ons/ctp/response/action/export/service/ActionExportServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  @Test
+  public void testacceptInstructionNullRequestAndNoCancel() {
+    // Given
+    ActionInstruction actionInstruction = new ActionInstruction();
+
+    ActionExportService actionExportService =
+        new ActionExportService(actionFeedbackPubl, mapperFacade, actionRequestRepo, addressRepo);
+
+    // Then
```
Comment (NickGrantX) [inline src/test/java/uk/gov/ons/ctp/response/action/export/service/ActionExportServiceTest.java:27]: This is more of a "when"

File: src/test/java/uk/gov/ons/ctp/response/action/export/service/ActionExportServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    actionExportService.acceptInstruction(actionInstruction);
+
+    assertTrue(true);
```
Comment (NickGrantX) [inline src/test/java/uk/gov/ons/ctp/response/action/export/service/ActionExportServiceTest.java:30]: This is the "then" and could just be a comment

**Review by NickGrantX — APPROVED** [review]
Very minor changes,... no reason not to approve. Good work 👍

File: src/main/resources/database/changes/release-23/changelog.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+databaseChangeLog:
+
+  - changeSet:
+      id: 23-1
+      author: Luke Loze
+      changes:
+        - sqlFile:
+            comment: remove all mappring and templates
```
Comment (hbrace) [inline src/main/resources/database/changes/release-23/changelog.yml:8]: mappring? Typo?

**Review by hbrace — APPROVED** [review]
Very minor comment

**Pull request merged**